bloom-housing/bloom
# feat: submit for approval flow

# Pull Request Template

## Issue Overview

This PR addresses [#307](https://github.com/metrotranscom/doorway/issues/307)


## Description

When listings approval is on:
* When a partner submits a listing for approval, they get a modal asking them to confirm. When submitted, the status of the listing changes to pending approval.

When the feature is both on or off:
* Now using the ui-seeds Tag component in order to be able to match the pill designs for listings approval (the CSS overrides worked great!) - I updated to the new seeds Tag everywhere we were consuming UIC Tag in partners so that it stays visually consistent (locations below).
-- Listing form pages (Tag is the listing status
-- Application table pages (Tag is also the listing status)
-- Individual application form (Tag is the submission status of the application)
-- Application flagged set resolution page (Tag is the resolution status of the set)
-- User edit form (Tag is the confirmed status of the user) - (previously this was a Tag size that was inaccessible / we no longer support it, so now it's a little bigger)

* We had duplicated the logic around toggling which pill to show for listings, so I consolidated that into a new helper file.

There was one instance of Tag within some old flagged set files I went to update to the seeds version to discover we weren't using the files at all, so I removed them.

## How Can This Be Tested/Reviewed?

Check out the new pill visually in the places it was updated.
Ensure if you have the environment variable on that as a partner/jurisdictional admin when you create a draft listing and submit it for approval that a modal pops that matches the designs. On submit the status should change to pending approval.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
